confignode/src/main/java/org/apache/iotdb/confignode/persistence/PartitionInfo.java

```diff
@@ -56,40 +68,81 @@ public class PartitionInfoPersistence {
   // TODO: Serialize and Deserialize
   private final DataPartition dataPartition;
 
-  public PartitionInfoPersistence() {
+  private PartitionInfo() {
+    this.regionAllocateLock = new ReentrantReadWriteLock();
+    this.regionReadWriteLock = new ReentrantReadWriteLock();
+    this.regionMap = new HashMap<>();
+
     this.schemaPartitionReadWriteLock = new ReentrantReadWriteLock();
-    this.dataPartitionReadWriteLock = new ReentrantReadWriteLock();
     this.schemaPartition =
         new SchemaPartition(
             ConfigNodeDescriptor.getInstance().getConf().getSeriesPartitionExecutorClass(),
             ConfigNodeDescriptor.getInstance().getConf().getSeriesPartitionSlotNum());
     this.schemaPartition.setSchemaPartitionMap(new HashMap<>());
+
+    this.dataPartitionReadWriteLock = new ReentrantReadWriteLock();
     this.dataPartition =
         new DataPartition(
             ConfigNodeDescriptor.getInstance().getConf().getSeriesPartitionExecutorClass(),
             ConfigNodeDescriptor.getInstance().getConf().getSeriesPartitionSlotNum());
     this.dataPartition.setDataPartitionMap(new HashMap<>());
   }
 
+  public int generateNextRegionGroupId() {
+    int result;
+    regionAllocateLock.writeLock().lock();
+    try {
+      result = nextRegionGroupId;
```

Comment:
if nextRegionGroupId  use AtomicLong,  will be not need lock...

Reply:
Fixed~